nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -479,14 +563,29 @@ in {
         nextcloud-setup = let
           c = cfg.config;
           writePhpArrary = a: "[${concatMapStringsSep "," (val: ''"${toString val}"'') a}]";
+          requiresReadSecretFunction = c.dbpassFile != null || c.objectstore.s3.enable;
+          objectstoreConfig = let s3 = c.objectstore.s3; in optionalString s3.enable '''objectstore' => [
```

Comment:
PHP doesn't really care about indentation and this is not only weird to read, but also highly ambiguous because `'''` inside a multi-line string is actually `''` in the evaluated result.